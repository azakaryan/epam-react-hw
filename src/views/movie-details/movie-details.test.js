import React from 'react';
import Button from '@material-ui/core/Button/Button';
import { shallow } from 'enzyme';
import expectEnzymify from 'expect';
import enzymify from 'expect-enzyme';
import MovieDetails from './movie-details';

expectEnzymify.extend(enzymify());
const onLoad = () => {};
const match = {
  params: { id: 12 },
};

describe('Movie Home Component', () => {
  it('should render correctly', () => {
    const component = shallow(<MovieDetails onLoad={onLoad} match={match} />);
    expect(component).toMatchSnapshot();
  });

  it('renders a Movie Home component', () => {
    const wrapper = shallow(<MovieDetails onLoad={onLoad} match={match} />);
    expectEnzymify(wrapper).toBeA('div');
  });

  it('pass id when load element', () => {
    let _id = 0;
    const onLoad = (id) => _id = id;
    expect(_id).toEqual(0);
    const wrapper = shallow(<MovieDetails onLoad={onLoad} match={match} />);
    expect(_id).toEqual(12);
    wrapper.setProps({ match });
    expect(_id).toEqual(12);

    wrapper.setProps({ match: { params: { id: 15 } } });
    expect(_id).toEqual(15);
  });

  it('find element, test goHome functionality', () => {
    let url = 'movie-details';
    const history = {
      push: () => {
        url = '/';
      },
    };
    const wrapper = shallow(<MovieDetails onLoad={onLoad} match={match} history={history} />);
    const btn = wrapper.find({ id: 'search-btn'});
    expectEnzymify(btn).toBeA(Button);

    btn.simulate('click');
    expect(url).toEqual('/');
  });
});
