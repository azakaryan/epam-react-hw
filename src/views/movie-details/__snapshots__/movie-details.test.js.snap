// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Home Component should render correctly 1`] = `
<div
  className="main"
  id="movie-details"
>
  <div
    className="search"
  >
    <WithStyles(Button)
      color="primary"
      id="search-btn"
      onClick={[Function]}
      variant="contained"
    >
      SEARCH
    </WithStyles(Button)>
  </div>
  <Connect(MovieCover) />
  <Connect(MovieList) />
</div>
`;
